src/ax/saga/persistence/eventsourcing/persister.go

```diff
@@ -142,10 +147,14 @@ func (w *unitOfWork) shouldSnapshot(before, after saga.Revision) bool {
 		return false
 	}
 
+	if before == 0 {
+		return true
+	}
+
 	freq := w.frequency
 	if freq == 0 {
 		freq = DefaultSnapshotFrequency
 	}
 
-	return (before / freq) != (after / freq)
+	return (after-before)%freq == 0
```

Comment:
I *think* this logic can just be `(after - before) >= freq`, then the need for `before == 0` check on line 150 is unnecessary.